### PR TITLE
Remove unused code from starkware_utils

### PR DESCRIPTION
**Pull request by GMKrieger**
Issue Number: N/A

## Type

- [ ] feature
- [ ] bugfix
- [x] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Description

Removed several unused methods and unreachable code from `starkware_utils`.

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

File: crates/starknet-os/src/starkware_utils/commitment_tree/binary_fact_tree_node.rs [hunk 1]
```diff
@@ -174,20 +174,7 @@ where
         ffc: &'a mut FactFetchingContext<S, H>,
         indices: &'a [TreeIndex],
         facts: &'a mut Option<BinaryFactDict>,
-    ) -> impl std::future::Future<Output = Result<HashMap<TreeIndex, LF>, TreeError>> + Send {
```
Comment (whichqua) [inline crates/starknet-os/src/starkware_utils/commitment_tree/binary_fact_tree_node.rs:177]: Any reason why you have removed this default implementation?

**Review by whichqua — COMMENTED** [review]
Left a single comment but most seems good.

File: crates/starknet-os/src/starkware_utils/commitment_tree/binary_fact_tree_node.rs (hunk 1 above)
Comment (GMKrieger) [reply crates/starknet-os/src/starkware_utils/commitment_tree/binary_fact_tree_node.rs:177]: No particular reason except it's never used. This trait is only implemented in `virtual_patricia_node.rs`, and it has it's own implementation of _get_leaves.

**Review by ftheirs — APPROVED** [review]
This looks good 💪

**Review by notlesh — COMMENTED** [review]
A lot of what is removed here was from the original port of the pythonic impl to rust. That was done in as much of a literal fashion as was possible, and that's probably why there is a lot of dead code.

I can see arguments for and against keeping this code, so perhaps we should discuss internally before we decide.

**Review by whichqua — APPROVED** [review]
LGTM!

**Review by HermanObst — APPROVED** [review]
LGTM

**Pull request merged**